### PR TITLE
K8s: Make Postgres hosting safer

### PR DESCRIPTION
**Pull request by lawrencegripper**
TLDR: The postgres hosting described may to corruption of the DB. This PR updates to try and make this less likely.

Ps. Not a postgres expert, did [some reading](https://stackoverflow.com/questions/75828221/how-to-safely-restart-postgresql-container-in-kubernetes/75829325#75829325) after corrupting my DB in k8s. 

## Why can this happen?

The current docs suggest hosting the `postgres` as a sidecar to the atuin container.

This means:
1. Updates to `atuin` results in restart of the `postgres` container. 
1. Crashes in `atuin` cause not only `atuin` to restart but also `postgres`
1. The `Strategy` for the `Deployment` default to [`RollingUpdate`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy) so a new pod is brought up before the old one is stopped

When updating atuin we therefor end up with two postgres instances, one for the old and one for the new, running in parallel. 

Additionally when stopping the old postgres container K8s there is no warning given to the postgres instance. 

I hit this issues with my setup. I think I made it more likely to occur as I a `hostPath` mount vs the shown `pvc` mounts.

## What's the change?

1. Split `atuin` from `postgres` into it's own `deployment` so updating atuin doesn't require restarting postgres. Likewise crashing `atuin` doesn't restart postgres.
2. Set `postgres` deployment to use `Recreate` strategy so old instance is stopped then new is started when updating to avoid 2 instances running.
3. Add a `pre-stop` hook to `postgres` which handles gracefully informing `postgres` it's about to be stopped before k8s stops it.
4. Add a `cronJob` example which is configured to take a backup of the DB on a schedule.

### CODE REVIEW

File: src/content/docs/self-hosting/kubernetes.mdx [hunk 1]
```diff
@@ -26,7 +26,7 @@ stringData:
   ATUIN_HOST: "127.0.0.1"
   ATUIN_PORT: "8888"
   ATUIN_OPEN_REGISTRATION: "true"
-  ATUIN_DB_URI: "postgres://atuin:seriously-insecure@localhost/atuin"
+  ATUIN_DB_URI: "postgres://atuin:seriously-insecure@postgres/atuin"
```
Comment (lawrencegripper) [inline src/content/docs/self-hosting/kubernetes.mdx:29]: Postgres is now accessible via the `postgres` service and no longer in the same pod.

File: src/content/docs/self-hosting/kubernetes.mdx [hunk 2]
```diff
@@ -131,6 +157,21 @@ spec:
   selector:
     io.kompose.service: atuin
 ---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    io.kompose.service: postgres
+  name: postgres
```
Comment (lawrencegripper) [inline src/content/docs/self-hosting/kubernetes.mdx:165]: This is the new service which is dns addressible from the atuin pod as `postgres`

File: src/content/docs/self-hosting/kubernetes.mdx [hunk 3]
```diff
@@ -186,6 +227,11 @@ metadata:
     name: atuin
 ```
 
+:::caution
+This configuration uses a `deployment` to host the PostgresDB.
+Consider using a `StatefulSet` instead as a 
+:::
+
```
Comment (lawrencegripper) [inline src/content/docs/self-hosting/kubernetes.mdx:234]:
```suggestion
```

**Review by ellie — APPROVED** [review]
Thank you! I'll merge this for now, but some context on my plans here: I'd like to direct people towards the helm chart, and not help them with postgres at all. Imo hosting postgres should be an exercise left to the reader, and if they're unable to do so they should consider using a managed service. It's pretty out of scope for our docs.

**Comment by lawrencegripper** [comment]
Yeah, that sounds like a good call to me 👍
